### PR TITLE
NeuralNDCG, 1.4.0 release

### PR DESCRIPTION
**Pull request by sadaharu-inugami**
This PR introduces NeuralNDCG, a loss function for direct optimisation of NDCG metric ([arxiv link](https://arxiv.org/abs/2102.07831)). The loss function is implemented in two variants - standard (approximating each position's gain) and transposed (approximating each document's discount).

Required Pytorch and torchvision versions are now ranges, 1.6.0-1.8.0 and 0.7.0-0.9.0, respectively.

Because of these relatively important changes, a minor version bump is done to 1.4.0 version.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,14 +17,15 @@ We hope that allRank will facilitate both research in neural LTR and its industr
...
- 1. ListNet (For a binary and graded relevance)
+ 1. ListNet (for binary and graded relevance)
  2. ListMLE
  3. RankNet
  4. Ordinal loss
  5. LambdaRank
  6. LambdaLoss
  7. ApproxNDCG
  8. RMSE
+ 9. NeuralNDCG (introduced in https://arxiv.org/pdf/2102.07831)
```
Comment (PrzemekPobrotyn) [inline README.md:28]: you could also add a note about NeuralNDCG paper to the Research section of readme

Comment (sadaharu-inugami) [reply README.md:28]: Added!

**Review by PrzemekPobrotyn — APPROVED** [review]

**Review by mhsyno — APPROVED** [review]

**Comment by gzls90** [comment]
Hey, could there be possibility to implement NeuralNDCG with improved sorting/ranking (paper https://arxiv.org/pdf/2002.08871.pdf by google researchers). Pytorch optimized implementation is already available in https://github.com/teddykoker/torchsort . They report N log(N) sorting speed without tradeoff, whereas original NeuralNDCG uses slower variant AFAIK. Speed is important factor for this objetive to work on large datasets IMO. WDYT?

**Comment by sadaharu-inugami** [comment]
> Hey, could there be possibility to implement NeuralNDCG with improved differentiable sorting/ranking (paper https://arxiv.org/pdf/2002.08871.pdf by google researchers). Pytorch optimized implementation is already available in https://github.com/teddykoker/torchsort . They report N log(N) sorting speed without tradeoff, whereas original NeuralNDCG uses slower variant AFAIK. Speed is important factor for this objetive to work on large datasets IMO. WDYT?

We will definitely consider that after releasing reproduction guides for both research papers created with AllRank. Thank you for this suggestion.
